### PR TITLE
AIRFLOW-1772: Fix bug with handling cron expressions as an schedule i…

### PR DESCRIPTION
**Pull request by samelamin**
Dear Airflow maintainers,

Please accept this PR. I understand that it will not be reviewed until I have checked off all the steps below!

### JIRA
- [ ] My PR addresses the following [Airflow JIRA](https://issues.apache.org/jira/browse/AIRFLOW-1772) issues and references them in the PR title. 
    - https://issues.apache.org/jira/browse/AIRFLOW-1772

### Description
- [ ] The schedule interval can be a timedelta OR a string, the sensor works fine when the interval is set to a time delta but dies when its a string because of [this](https://github.com/apache/incubator-airflow/blob/master/airflow/contrib/sensors/gcs_sensor.py#L70)

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -133,6 +133,7 @@ def write_version(filename=os.path.join(*['airflow',
     'oauth2client>=2.0.2, <2.1.0',
     'PyOpenSSL',
     'google-cloud-dataflow',
+    'croniter',
```
Comment (bolkedebruin) [inline setup.py]: this is redundant

File: airflow/contrib/sensors/gcs_sensor.py [hunk 2]
```diff
@@ -67,6 +69,10 @@ def ts_function(context):
     behaviour is check for the object being updated after execution_date +
     schedule_interval.
     """
+    schedule_interval = context['dag'].schedule_interval
+    if isinstance(schedule_interval, str):
+        iterator = croniter(schedule_interval, context['execution_date'])
+        return iterator.get_next(datetime)
```
Comment (bolkedebruin) [inline airflow/contrib/sensors/gcs_sensor.py]: if this current date time (not initialised here it seems), use datetime.now(pytz.utc)

(yes pytz needs to be added)

**Comment by bolkedebruin** [comment]
Please verify your commit message against the requirements and add tests please.

**Comment by samelamin** [comment]
I have refactored the commits based on your comments

However I do not see any reference to tests for these classes, if it helps I refactored the PR to use the dag's following_schedule for intervals as seen [here](https://github.com/apache/incubator-airflow/blob/a81c153cc48e4c99a9e0a5047990b84c5d07e3cb/airflow/operators/latest_only_operator.py)

File: airflow/contrib/sensors/gcs_sensor.py [hunk 3]
```diff
@@ -15,7 +15,6 @@
 from airflow.operators.sensors import BaseSensorOperator
 from airflow.utils.decorators import apply_defaults
 
-
```
Comment (skudriashev) [inline airflow/contrib/sensors/gcs_sensor.py:18]: Why removed? Need to keep 2 empty lines before class declaration (pep8).

File: airflow/contrib/sensors/gcs_sensor.py [hunk 4]
```diff
@@ -67,8 +66,7 @@ def ts_function(context):
     behaviour is check for the object being updated after execution_date +
     schedule_interval.
     """
-    return context['execution_date'] + context['dag'].schedule_interval
-
```
Comment (skudriashev) [inline airflow/contrib/sensors/gcs_sensor.py:71]: Same here.

**Review by skudriashev — CHANGES_REQUESTED** [review]

File: airflow/contrib/sensors/gcs_sensor.py (hunk 3 above)
Comment (samelamin) [reply airflow/contrib/sensors/gcs_sensor.py:18]: i can change it but is this the only change required ? If there is anything else I would rather know now so I can change it once as this PR has been open for quite a while

**Comment by ifuller1** [comment]
@samelamin - thanks for raising this PR as I was also caught out by this.

@skudriashev @bolkedebruin - any chance you could help @samelamin get his PR merged? The contribution will be valid for others.

File: airflow/contrib/sensors/gcs_sensor.py [hunk 5]
```diff
@@ -67,8 +66,7 @@ def ts_function(context):
...
+    return context['dag'].following_schedule(context['execution_date'])
```
Comment (ashb) [inline airflow/contrib/sensors/gcs_sensor.py:69]: Isn't this already in the context as `context['next_execution_date']`?

**Pull request closed without merging**